File: chatuikit/src/main/java/com/cometchat/chatuikit/shared/spans/StrikethroughFormatSpan.java
package com.cometchat.chatuikit.shared.spans;

import android.text.style.StrikethroughSpan;

import com.cometchat.chatuikit.shared.views.richtexttoolbar.FormatType;

/**
 * Span for strikethrough formatting in WYSIWYG rich text editing.
 * <p>
 * This span extends {@link StrikethroughSpan} to render text with a strikethrough
 * line. It implements {@link RichTextFormatSpan} to enable format detection
 * and markdown conversion.
 * </p>
 * <p>
 * When applied to text, this span displays the text with a strikethrough line
 * without showing any markdown syntax markers (like ~~). The markdown markers
 * are only generated when converting to markdown for sending messages.
 * </p>
 *
 * @see RichTextFormatSpan
 * @see FormatType#STRIKETHROUGH
 */
public class StrikethroughFormatSpan extends StrikethroughSpan implements RichTextFormatSpan {

    /**
     * Creates a new StrikethroughFormatSpan that renders text with strikethrough style.
     */
    public StrikethroughFormatSpan() {
        super();
    }

    /**
     * Returns the format type associated with this span.
     *
     * @return {@link FormatType#STRIKETHROUGH} indicating this is a strikethrough format span.
     */
    @Override
    public FormatType getFormatType() {
        return FormatType.STRIKETHROUGH;
    }
}

File: chatuikit-kotlin/src/main/java/com/cometchat/uikit/kotlin/shared/spans/StrikethroughFormatSpan.kt
package com.cometchat.uikit.kotlin.shared.spans

import android.text.style.StrikethroughSpan
import com.cometchat.uikit.core.formatter.RichTextFormat

/**
 * Span for strikethrough formatting in WYSIWYG rich text editing.
 *
 * Extends [StrikethroughSpan] to render text with a strikethrough line.
 * Implements [RichTextFormatSpan] to enable format detection and markdown conversion.
 *
 * @see RichTextFormatSpan
 * @see RichTextFormat.STRIKETHROUGH
 */
class StrikethroughFormatSpan : StrikethroughSpan(), RichTextFormatSpan {

    override fun getFormatType(): RichTextFormat = RichTextFormat.STRIKETHROUGH
}
